packages/web-service/src/pages/habitat/a24/sett/common/date/dates.js

```diff
@@ -0,0 +1,24 @@
+import { APIRequests } from '../../../../../../services/api-requests.js'
+import { A24_SETT } from '../../../../../tasklist/a24-badger-licence.js'
+import { isCompleteOrConfirmed } from '../../../../../common/tag-functions.js'
+import { habitatURIs } from '../../../../../../uris.js'
+
+export const getDateData = (dateToGet) => {
+  if (dateToGet) {
+    const date = new Date(dateToGet)
+    return {
+      year: date.getFullYear(),
+      month: date.getMonth() + 1,
+      day: date.getDate()
+    }
+  }
+  return null
+}
+
+export const dateCompletion = async (journeyData, uri) => {
```

Comment:
alot of the completions follow the same pattern of
```
export const completion = async request => {
  const journeyData = await request.cache().getData()
  const tagState = await APIRequests.APPLICATION.tags(journeyData.applicationId).get(A24_SETT)

  if (isCompleteOrConfirmed(tagState)) {
    return habitatURIs.CHECK_YOUR_ANSWERS.uri
  }
  return habitatURIs.GRID_REF.uri
}
```

we can make this generic and just pass in the return URI as above as this is handling the checkAnswers functionality

Reply:
This looks like it can be generic with passing in the appId, .get(FIELD), checkAnswerURI, nextPageURI